tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Common/MatterProvision.cs

```diff
@@ -1156,25 +1160,116 @@ private static string TeamMembersPermissionInformation(MatterDetails matterDetai
         /// <returns>returns dictionary object</returns>
         internal Dictionary<string, string> SetStampProperty(Client client, Matter matter, MatterDetails matterDetails)
         {
+            string documentTemplateCount = string.Join(ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR, matter.DocumentTemplateCount);
             string matterCenterPermission = string.Join(ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR, matter.Permissions);
             string matterCenterRoles = string.Join(ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR, matter.Roles);
-            string documentTemplateCount = string.Join(ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR, matter.DocumentTemplateCount);
+            string finalTeamMembers = matterDetails.TeamMembers;
+            string finalResponsibleAttorneysUsers = matterDetails.ResponsibleAttorney;
+            string finalResponsibleAttorneysEmail = matterDetails.ResponsibleAttorneyEmail;
             string matterCenterUsers = string.Empty;
             string matterCenterUserEmails = string.Empty;
             string separator = string.Empty;
             foreach (IList<string> userNames in matter.AssignUserNames)
             {
-                matterCenterUsers += separator + string.Join(ServiceConstants.SEMICOLON, userNames.Where(user => !string.IsNullOrWhiteSpace(user)));
-                separator = ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR;
+                if (userDetails.CheckUserPresentInMatterCenter(client.Url, userNames.Where(user => !string.IsNullOrWhiteSpace(user)).SingleOrDefault()) == true)
+                {
+                    matterCenterUsers += separator + string.Join(ServiceConstants.SEMICOLON, userNames.Where(user => !string.IsNullOrWhiteSpace(user)));
+                    separator = ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR;
+                }
             }
 
             foreach (IList<string> userEmails in matter.AssignUserEmails)
             {
-                matterCenterUserEmails += string.Join(ServiceConstants.SEMICOLON, userEmails.Where(user => !string.IsNullOrWhiteSpace(user))) + separator;
+                if (userDetails.CheckUserPresentInMatterCenter(client.Url, userEmails.Where(user => !string.IsNullOrWhiteSpace(user)).SingleOrDefault()) == true)
+                {
+                    matterCenterUserEmails += string.Join(ServiceConstants.SEMICOLON, userEmails.Where(user => !string.IsNullOrWhiteSpace(user))) + separator;
+                }
+            }           
+            var finalMatterPermissionsList = matterCenterPermission.Replace("$|$", "$").Split('$').ToList();

```

Comment:
In matter creation, when the matter object contains both internal and external users, in the property bag, we have to store only the internal users until the external user accepts the invitation

First remove the external user, his roles, his permissions from the matter object and update the property bag with just internal users

After storing the internal user and his roles and permissions in the property bag, remove the internal user from the matter object and store that matter object with just external user in the table storage